crates/subspace-networking/src/node_runner.rs

```diff
@@ -411,6 +422,23 @@ where
                 };
                 debug!("Connection closed with peer {peer_id} [{num_established} from peer]");
 
+                // TODO: Workaround for https://github.com/libp2p/rust-libp2p/discussions/3418
```

Comment:
Can we get rid of `shared.connected_peers_count` and sue this map (this is much better!)?

Reply:
As mentioned in PR description, the idea is to get to the bottom of it and simply delete all the code added in this PR like it never existed.